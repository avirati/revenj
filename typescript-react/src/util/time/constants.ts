import { unitOfTime } from 'moment';

// Used for moment.js time comparison in the validations. Do not change the strings.
export const UNIT_MILLISECOND: unitOfTime.DurationConstructor = 'millisecond';
export const UNIT_SECOND: unitOfTime.DurationConstructor = 'second';
export const UNIT_MINUTE: unitOfTime.DurationConstructor = 'minute';
export const UNIT_HOUR: unitOfTime.DurationConstructor = 'hour';
export const UNIT_DAY: unitOfTime.DurationConstructor = 'day';
export const UNIT_WEEK: unitOfTime.DurationConstructor = 'week';
export const UNIT_MONTH: unitOfTime.DurationConstructor = 'month';
export const UNIT_YEAR: unitOfTime.DurationConstructor = 'year';
