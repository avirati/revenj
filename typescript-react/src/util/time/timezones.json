{
	"version": "2017b",
	"zones": [
		"Africa/Accra|LMT GMT +0020|.Q 0 -k|012121212121212121212121212121212121212121212121|-26BbX.8 6tzX.8 MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE|41e5",
    "Africa/Lagos|LMT WAT|-d.A -10|01|-22y0d.A|17e6",
    "Africa/Nairobi|LMT EAT +0230 +0245|-2r.g -30 -2u -2J|01231|-1F3Cr.g 3Dzr.g okMu MFXJ|47e5",
		"Africa/Maputo|LMT CAT|-2a.k -20|01|-2GJea.k|26e5",
		"Africa/Johannesburg|SAST SAST SAST|-1u -20 -30|012121|-2GJdu 1Ajdu 1cL0 1cN0 1cL0|84e5",
		"Africa/Monrovia|MMT MMT GMT|H.8 I.u 0|012|-23Lzg.Q 28G01.m|11e5",
		"America/Port-au-Prince|PPMT EST EDT|4N 50 40|01212121212121212121212121212121212121212121212121212121212121212121212121212121212121|-28RHb 2FnMb 19X0 1cN0 1cL0 1cN0 1cL0 1cN0 1cL0 1cN0 1cL0 14q0 1o00 11A0 1o00 11A0 1o00 14o0 1lc0 14o0 1lc0 14o0 1o00 11A0 1o00 11A0 1o00 14o0 1lc0 14o0 1lc0 i6n0 1nX0 11B0 1nX0 d430 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 3iN0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0|23e5",
    "Asia/Manila|+08 +09|-80 -90|010101010|-1kJI0 AL0 cK10 65X0 mXB0 vX0 VK10 1db0|24e6",
    "Asia/Rangoon|RMT +0630 +09|-6o.L -6u -90|0121|-21Jio.L SmnS.L 7j9u|48e5",
    "EET|EET EEST|-20 -30|010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010|hDB0 1a00 1fA0 1cM0 1cM0 1cM0 1fA0 1a00 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1fA0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1fA0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1fA0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00",
		"EST|EST|50|0|",
		"EST5EDT|EST EDT EWT EPT|50 40 40 40|010102301010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010|-261t0 1nX0 11B0 1nX0 SgN0 8x40 iv0 QwN0 1cL0 1cN0 1cL0 1cN0 1cL0 1cN0 1cL0 1cN0 1fz0 1cN0 1cL0 1cN0 1cL0 s10 1Vz0 LB0 1BX0 1cN0 1fz0 1a10 1fz0 1cN0 1cL0 1cN0 1cL0 1cN0 1cL0 1cN0 1cL0 1cN0 1fz0 1a10 1fz0 1cN0 1cL0 1cN0 1cL0 1cN0 1cL0 14p0 1lb0 14p0 1nX0 11B0 1nX0 11B0 1nX0 14p0 1lb0 14p0 1lb0 14p0 1nX0 11B0 1nX0 11B0 1nX0 14p0 1lb0 14p0 1lb0 14p0 1lb0 14p0 1nX0 11B0 1nX0 11B0 1nX0 14p0 1lb0 14p0 1lb0 14p0 1nX0 11B0 1nX0 11B0 1nX0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Rd0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0 Op0 1zb0",
		"Europe/Belgrade|CET CEST|-10 -20|01010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010|-19RC0 3IP0 WM0 1fA0 1cM0 1cM0 1rc0 Qo0 1vmo0 1cM0 1cM0 1fA0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1fA0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1cM0 1fA0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00|12e5",
		"Europe/Dublin|DMT IST GMT BST IST|p.l -y.D 0 -10 -10|01232323232324242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242424242|-2ax9y.D Rc0 1fzy.D 14M0 1fc0 1g00 1co0 1dc0 1co0 1oo0 1400 1dc0 19A0 1io0 1io0 WM0 1o00 14o0 1o00 17c0 1io0 17c0 1fA0 1a00 1lc0 17c0 1io0 17c0 1fA0 1a00 1io0 17c0 1io0 17c0 1fA0 1cM0 1io0 17c0 1fA0 1a00 1io0 17c0 1io0 17c0 1fA0 1a00 1io0 1qM0 Dc0 g5X0 14p0 1wn0 17d0 1io0 11A0 1o00 17c0 1fA0 1a00 1fA0 1cM0 1fA0 1a00 17c0 1fA0 1a00 1io0 17c0 1lc0 17c0 1fA0 1a00 1io0 17c0 1io0 17c0 1fA0 1a00 1a00 1qM0 WM0 1qM0 11A0 1o00 WM0 1qM0 WM0 1qM0 WM0 1qM0 WM0 1tA0 IM0 90o0 U00 1tA0 U00 1tA0 U00 1tA0 U00 1tA0 WM0 1qM0 WM0 1qM0 WM0 1tA0 U00 1tA0 U00 1tA0 11z0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1o00 14o0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00 11A0 1qM0 WM0 1qM0 WM0 1qM0 WM0 1qM0 11A0 1o00 11A0 1o00|12e5",
		"Etc/GMT+0|GMT|0|0|",
		"Etc/GMT+1|-01|10|0|",
		"Etc/GMT+10|-10|a0|0|",
		"Etc/GMT+11|-11|b0|0|",
		"Etc/GMT+12|-12|c0|0|",
		"Etc/GMT+3|-03|30|0|",
		"Etc/GMT+4|-04|40|0|",
		"Etc/GMT+5|-05|50|0|",
		"Etc/GMT+6|-06|60|0|",
		"Etc/GMT+7|-07|70|0|",
		"Etc/GMT+8|-08|80|0|",
		"Etc/GMT+9|-09|90|0|",
		"Etc/GMT-1|+01|-10|0|",
		"Etc/GMT-13|+13|-d0|0|",
		"Etc/GMT-14|+14|-e0|0|",
		"Etc/GMT-2|+02|-20|0|",
		"Etc/GMT-3|+03|-30|0|",
		"Etc/GMT-4|+04|-40|0|",
		"Etc/GMT-5|+05|-50|0|",
		"Etc/GMT-6|+06|-60|0|",
		"Etc/GMT-8|+08|-80|0|",
		"Etc/UCT|UCT|0|0|",
		"Etc/UTC|UTC|0|0|"
	],
	"links": [
    "Etc/GMT+0|Etc/GMT",
		"Etc/GMT+0|Etc/GMT-0",
		"Etc/GMT+0|Etc/GMT0",
		"Etc/GMT+0|Etc/Greenwich",
		"Etc/GMT+0|GMT",
		"Etc/GMT+0|GMT+0",
		"Etc/GMT+0|GMT-0",
		"Etc/GMT+0|GMT0",
		"Etc/GMT+0|Greenwich",
		"Etc/UCT|UCT",
		"Etc/UTC|Etc/Universal",
		"Etc/UTC|Etc/Zulu",
		"Etc/UTC|UTC",
		"Etc/UTC|Universal",
		"Etc/UTC|Zulu",
		"Africa/Abidjan|Africa/Bamako",
		"Africa/Abidjan|Africa/Banjul",
		"Africa/Abidjan|Africa/Conakry",
		"Africa/Abidjan|Africa/Dakar",
		"Africa/Abidjan|Africa/Freetown",
		"Africa/Abidjan|Africa/Lome",
		"Africa/Abidjan|Africa/Ouagadougou",
		"Africa/Abidjan|Africa/Sao_Tome",
		"Africa/Abidjan|Africa/Timbuktu",
		"Africa/Abidjan|Atlantic/St_Helena",
		"Africa/Cairo|Egypt",
		"Africa/Johannesburg|Africa/Maseru",
		"Africa/Johannesburg|Africa/Mbabane",
		"Africa/Khartoum|Africa/Juba",
		"Africa/Lagos|Africa/Bangui",
		"Africa/Lagos|Africa/Brazzaville",
		"Africa/Lagos|Africa/Douala",
		"Africa/Lagos|Africa/Kinshasa",
		"Africa/Lagos|Africa/Libreville",
		"Africa/Lagos|Africa/Luanda",
		"Africa/Lagos|Africa/Malabo",
		"Africa/Lagos|Africa/Niamey",
		"Africa/Lagos|Africa/Porto-Novo",
		"Africa/Maputo|Africa/Blantyre",
		"Africa/Maputo|Africa/Bujumbura",
		"Africa/Maputo|Africa/Gaborone",
		"Africa/Maputo|Africa/Harare",
		"Africa/Maputo|Africa/Kigali",
		"Africa/Maputo|Africa/Lubumbashi",
		"Africa/Maputo|Africa/Lusaka",
		"Africa/Nairobi|Africa/Addis_Ababa",
		"Africa/Nairobi|Africa/Asmara",
		"Africa/Nairobi|Africa/Asmera",
		"Africa/Nairobi|Africa/Dar_es_Salaam",
		"Africa/Nairobi|Africa/Djibouti",
		"Africa/Nairobi|Africa/Kampala",
		"Africa/Nairobi|Africa/Mogadishu",
		"Africa/Nairobi|Indian/Antananarivo",
		"Africa/Nairobi|Indian/Comoro",
		"Africa/Nairobi|Indian/Mayotte",
    "Africa/Tripoli|Libya",
    "Asia/Rangoon|Asia/Yangon",
		"Europe/Belgrade|Europe/Zagreb"
	]
}
