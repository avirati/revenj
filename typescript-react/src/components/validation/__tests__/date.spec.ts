import {
  UNIT_DAY,
  UNIT_MONTH,
} from '../../../util/time/constants';
import { time } from '../../../util/time/time';
import {
  afterTheDateCreator,
  maxInTheFutureCreator,
  maxInThePastCreator,
  notInTheFutureCreator,
  notInThePastCreator,
  sameAsOrAfterFieldCreator,
  sameAsOrAfterTheDateCreator,
  sameAsOrBeforeFieldCreator,
  sameAsOrBeforeTheDateCreator,
} from '../date';

interface IDateForm {
  before: DateStr;
  after: DateStr;
}

const mockDateForm: IDateForm = {
  after: '2018-03-14' as DateStr,
  before: '2018-03-14' as DateStr,
};

describe('validation', () => {
  describe('notInTheFuture', () => {
    const notInTheFuture = notInTheFutureCreator(UNIT_DAY)();

    it('should return undefined if the input date is not in the future', () => {
      expect(notInTheFuture(time().format() as DateStr)).toBeUndefined();
      expect(notInTheFuture(time().subtract(1, UNIT_DAY).format() as DateStr)).toBeUndefined();
      expect(notInTheFuture(undefined as any)).toBeUndefined();
    });

    it('should return an error message if the input date is in the future', () => {
      expect(notInTheFuture(time().add(2, UNIT_DAY).format() as DateStr)).toBeDefined();
      expect(notInTheFuture(time().add(2, UNIT_DAY).format() as DateStr)).toMatchSnapshot();
    });
  });

  describe('notInThePast', () => {
    const notInThePast = notInThePastCreator(UNIT_DAY)();

    it('should return undefined if the input date is not in the past', () => {
      expect(notInThePast(time().format() as DateStr)).toBeUndefined();
      expect(notInThePast(time().add(1, UNIT_DAY).format() as DateStr)).toBeUndefined();
      expect(notInThePast(undefined as any)).toBeUndefined();
    });

    it('should return an error message if the input date is in the past', () => {
      expect(notInThePast(time().subtract(2, UNIT_DAY).format() as DateStr)).toBeDefined();
      expect(notInThePast(time().subtract(2, UNIT_DAY).format() as DateStr)).toMatchSnapshot();
    });
  });

  describe('afterTheDateCreator', () => {
    const afterTheDate = afterTheDateCreator({
      date: '2018-03-14' as DateStr,
    })();

    it('should return undefined if the input date is after some date', () => {
      expect(afterTheDate('2018-03-15' as DateStr)).toBeUndefined();
      expect(afterTheDate(undefined as any)).toBeUndefined();
    });

    it('should return an error message if the input date is before or same as some date', () => {
      expect(afterTheDate('2018-03-13' as DateStr)).toBeDefined();
      expect(afterTheDate('2018-03-13' as DateStr)).toMatchSnapshot();
      expect(afterTheDate('2018-03-14' as DateStr)).toBeDefined();
      expect(afterTheDate('2018-03-14' as DateStr)).toMatchSnapshot();
      expect(afterTheDate('2018-03-14T00:00:01' as DateStr)).toBeDefined();
      expect(afterTheDate('2018-03-14T00:00:01' as DateStr)).toMatchSnapshot();
      expect(afterTheDate('2018-03-13T23:59:59' as DateStr)).toBeDefined();
      expect(afterTheDate('2018-03-13T23:59:59' as DateStr)).toMatchSnapshot();
    });
  });

  describe('sameAsOrAfterTheDateCreator', () => {
    const sameAsOrAfterTheDate = sameAsOrAfterTheDateCreator({
      date: '2018-03-14' as DateStr,
    })();

    it('should return undefined if the input date is on or after some date', () => {
      expect(sameAsOrAfterTheDate('2018-03-14' as DateStr)).toBeUndefined();
      expect(sameAsOrAfterTheDate('2018-03-14T00:00:01' as DateStr)).toBeUndefined();
      expect(sameAsOrAfterTheDate('2018-03-15' as DateStr)).toBeUndefined();
      expect(sameAsOrAfterTheDate(undefined as any)).toBeUndefined();
    });

    it('should return an error message if the input date is before some date', () => {
      expect(sameAsOrAfterTheDate('2018-03-13' as DateStr)).toBeDefined();
      expect(sameAsOrAfterTheDate('2018-03-13' as DateStr)).toMatchSnapshot();
      expect(sameAsOrAfterTheDate('2018-03-13T23:59:59' as DateStr)).toBeDefined();
      expect(sameAsOrAfterTheDate('2018-03-13T23:59:59' as DateStr)).toMatchSnapshot();
    });
  });

  describe('sameAsOrBeforeTheDateCreator', () => {
    const sameAsOrBeforeTheDate = sameAsOrBeforeTheDateCreator({
      date: '2018-03-14' as DateStr,
    })();

    it('should return undefined if the input date is on or before some date', () => {
      expect(sameAsOrBeforeTheDate('2018-03-14' as DateStr)).toBeUndefined();
      expect(sameAsOrBeforeTheDate('2018-03-14T23:59:59' as DateStr)).toBeUndefined();
      expect(sameAsOrBeforeTheDate('2018-03-13' as DateStr)).toBeUndefined();
      expect(sameAsOrBeforeTheDate(undefined as any)).toBeUndefined();
    });

    it('should return an error message if the input date is after some date', () => {
      expect(sameAsOrBeforeTheDate('2018-03-15' as DateStr)).toBeDefined();
      expect(sameAsOrBeforeTheDate('2018-03-15' as DateStr)).toMatchSnapshot();
      expect(sameAsOrBeforeTheDate('2018-03-15T23:59:59' as DateStr)).toBeDefined();
      expect(sameAsOrBeforeTheDate('2018-03-15T23:59:59' as DateStr)).toMatchSnapshot();
    });
  });

  describe('sameAsOrAfterFieldCreator', () => {
    const sameAsOrAfter = sameAsOrAfterFieldCreator<IDateForm>(
      { label: 'Date', path: 'date' },
      { label: 'After', path: 'after' },
    )();

    it('should return undefined if input date is after or same as field date', () => {
      expect(sameAsOrAfter('2018-03-14' as DateStr, mockDateForm)).toBeUndefined();
      expect(sameAsOrAfter('2018-03-15' as DateStr, mockDateForm)).toBeUndefined();
      expect(sameAsOrAfter(undefined as any, mockDateForm)).toBeUndefined();
    });

    it('should return an error message if input date is before the field date', () => {
      expect(sameAsOrAfter('2018-03-13' as DateStr, mockDateForm)).toBeDefined();
      expect(sameAsOrAfter('2018-03-13' as DateStr, mockDateForm)).toMatchSnapshot();
    });
  });

  describe('sameAsOrBeforeFieldCreator', () => {
    const sameAsOrBefore = sameAsOrBeforeFieldCreator<IDateForm>(
      { label: 'Date', path: 'date' },
      { label: 'Before', path: 'before' },
    )();

    it('should return undefined if input date is after or same as field date', () => {
      expect(sameAsOrBefore('2018-03-14' as DateStr, mockDateForm)).toBeUndefined();
      expect(sameAsOrBefore('2018-03-13' as DateStr, mockDateForm)).toBeUndefined();
      expect(sameAsOrBefore(undefined as any, mockDateForm)).toBeUndefined();
    });

    it('should return an error message if input date is before the field date', () => {
      expect(sameAsOrBefore('2018-03-15' as DateStr, mockDateForm)).toBeDefined();
      expect(sameAsOrBefore('2018-03-15' as DateStr, mockDateForm)).toMatchSnapshot();
    });
  });

  describe('maxInTheFutureCreator', () => {
    const max10DaysInFuture = maxInTheFutureCreator(10)();

    it('should return undefined if the date is within boundaries from today (default unit day)', () => {
      expect(max10DaysInFuture(time().format() as DateStr, {})).toBeUndefined();
      expect(max10DaysInFuture(time().subtract(1, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
      expect(max10DaysInFuture(time().add(1, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
      expect(max10DaysInFuture(time().add(10, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
    });

    it('should return an error message if the input date is outside boundaries from today (default unit day)', () => {
      expect(max10DaysInFuture(time().add(11, UNIT_DAY).format() as DateStr, {})).toBe('Cannot be more than 10 days in the future');
    });

    it('should allow for customisation of precision/step', () => {
      const max1MonthInTheFuture = maxInTheFutureCreator(1, UNIT_MONTH)();
      expect(max1MonthInTheFuture(time().add(11, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
      expect(max1MonthInTheFuture(time().add(101, UNIT_DAY).format() as DateStr, {})).toBe('Cannot be more than 1 months in the future');
    });
  });

  describe('maxInThePastCreator', () => {
    const max10DaysInFuture = maxInThePastCreator(10)();

    it('should return undefined if the date is within boundaries from today (default unit day)', () => {
      expect(max10DaysInFuture(time().format() as DateStr, {})).toBeUndefined();
      expect(max10DaysInFuture(time().subtract(1, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
      expect(max10DaysInFuture(time().add(1, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
      expect(max10DaysInFuture(time().subtract(10, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
    });

    it('should return an error message if the input date is outside boundaries from today (default unit day)', () => {
      expect(max10DaysInFuture(time().subtract(11, UNIT_DAY).format() as DateStr, {})).toBe('Cannot be more than 10 days in the past');
    });

    it('should allow for customisation of precision/step', () => {
      const max1MonthInTheFuture = maxInThePastCreator(1, UNIT_MONTH)();
      expect(max1MonthInTheFuture(time().subtract(11, UNIT_DAY).format() as DateStr, {})).toBeUndefined();
      expect(max1MonthInTheFuture(time().subtract(101, UNIT_DAY).format() as DateStr, {})).toBe('Cannot be more than 1 months in the past');
    });
  });
});
