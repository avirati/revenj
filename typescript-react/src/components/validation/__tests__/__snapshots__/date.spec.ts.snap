// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation afterTheDateCreator should return an error message if the input date is before or same as some date 1`] = `"Date must be after 14 Mar 2018"`;

exports[`validation afterTheDateCreator should return an error message if the input date is before or same as some date 2`] = `"Date must be after 14 Mar 2018"`;

exports[`validation afterTheDateCreator should return an error message if the input date is before or same as some date 3`] = `"Date must be after 14 Mar 2018"`;

exports[`validation afterTheDateCreator should return an error message if the input date is before or same as some date 4`] = `"Date must be after 14 Mar 2018"`;

exports[`validation notInTheFuture should return an error message if the input date is in the future 1`] = `"Date can't be set in the future."`;

exports[`validation notInThePast should return an error message if the input date is in the past 1`] = `"Date can't be set in the past."`;

exports[`validation sameAsOrAfterFieldCreator should return an error message if input date is before the field date 1`] = `"Date must be same as or after After"`;

exports[`validation sameAsOrAfterTheDateCreator should return an error message if the input date is before some date 1`] = `"Date must be after or same as 14 Mar 2018"`;

exports[`validation sameAsOrAfterTheDateCreator should return an error message if the input date is before some date 2`] = `"Date must be after or same as 14 Mar 2018"`;

exports[`validation sameAsOrBeforeFieldCreator should return an error message if input date is before the field date 1`] = `"Date must be same as or before Before"`;

exports[`validation sameAsOrBeforeTheDateCreator should return an error message if the input date is after some date 1`] = `"Date must be before or same as 14 Mar 2018"`;

exports[`validation sameAsOrBeforeTheDateCreator should return an error message if the input date is after some date 2`] = `"Date must be before or same as 14 Mar 2018"`;
